src/backend/api/Fusion.Resources.Database/Entities/DbExternalPersonnelPerson.cs

```diff
@@ -37,15 +37,19 @@ public class DbExternalPersonnelPerson
         [MaxLength(100)]
         public string? LinkedInProfile { get; set; }
 
+        public bool IsDeleted { get; set; }
         public ICollection<DbPersonnelDiscipline> Disciplines { get; set; } = null!;
 
         internal static void OnModelCreating(ModelBuilder modelBuilder)
         {
             modelBuilder.Entity<DbExternalPersonnelPerson>(entity =>
             {
                 entity.Property(e => e.AccountStatus).HasConversion(new EnumToStringConverter<DbAzureAccountStatus>());
+                
                 entity.HasMany(e => e.Disciplines).WithOne().OnDelete(DeleteBehavior.Cascade);
                 entity.HasIndex(e => e.Mail).IsClustered(false);
+
+                entity.HasQueryFilter(e => !e.IsDeleted);
```

Comment:
How will this affect other entities which assumes person is not null? I.e. `DbResourceAllocationRequest.CreatedBy`

Reply:
This is ExternalPersons (not the one used for resource allocation requests DbPersons)
Where used, the row will just not appear.

That said, the rows should be available even though person is removed.
Will be removed.

There is a indicator in the contract personell list that should be sufficient